src/libs/actions/Travel.ts

```diff
@@ -1,59 +1,168 @@
+            value: {
                 isLoading: true,
+                errors: null,
             },
         },
     ];
 
-    const finallyData: OnyxUpdate[] = [
+    const successData: OnyxUpdate[] = [
         {
             onyxMethod: 'merge',
-            key: ONYXKEYS.NVP_TRAVEL_SETTINGS,
+            key: ONYXKEYS.TRAVEL_PROVISIONING,
             value: {
                 isLoading: false,
             },
         },
     ];
 
-    const error = new Error('Failed to generate spotnana token.');
-
-    return new Promise((resolve, reject) => {
-        asyncOpenURL(
-            // eslint-disable-next-line rulesdir/no-api-side-effects-method
-            API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.ACCEPT_SPOTNANA_TERMS, null, {optimisticData, finallyData})
-                .then((response) => {
-                    if (!response?.spotnanaToken) {
-                        reject(error);
-                        throw error;
-                    }
-                    return buildTravelDotURL(response.spotnanaToken);
-                })
-                .catch(() => {
-                    reject(error);
-                    throw error;
-                }),
-            (travelDotURL) => {
-                resolve(travelDotURL);
-                return travelDotURL ?? '';
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: 'merge',
+            key: ONYXKEYS.TRAVEL_PROVISIONING,
+            value: {
+                isLoading: false,
+                errors: getMicroSecondOnyxErrorWithTranslationKey('travel.errorMessage'),
             },
-        );
-    });
+        },
+    ];
+
+    const params: AcceptSpotnanaTermsParams = {domain};
+
+    API.write(WRITE_COMMANDS.ACCEPT_SPOTNANA_TERMS, params, {optimisticData, successData, failureData});
+}
+
+function handleProvisioningPermissionDeniedError(domain: string) {
+    Navigation.navigate(ROUTES.TRAVEL_DOMAIN_PERMISSION_INFO.getRoute(domain));
+    Onyx.merge(ONYXKEYS.TRAVEL_PROVISIONING, null);
+}
+
+function openTravelDotAfterProvisioning(spotnanaToken: string) {
+    Navigation.goBack();
+    Onyx.merge(ONYXKEYS.TRAVEL_PROVISIONING, null);
+    Linking.openURL(buildTravelDotURL(spotnanaToken));
+}
+
+function provisionDomain(domain: string) {
+    Navigation.goBack();
+    Onyx.merge(ONYXKEYS.TRAVEL_PROVISIONING, null);
+    Navigation.navigate(ROUTES.TRAVEL_TCS.getRoute(domain));
+}
+
+function bookATrip(translate: LocaleContextProps['translate'], setCtaErrorMessage: Dispatch<SetStateAction<string>>, ctaErrorMessage = ''): void {
+    if (!activePolicyID) {
+        return;
+    }
+    if (Str.isSMSLogin(primaryLogin)) {
+        setCtaErrorMessage(translate('travel.phoneError'));
+        return;
+    }
+    const policy = getPolicy(activePolicyID);
+    if (isEmptyObject(policy?.address)) {
+        Navigation.navigate(ROUTES.WORKSPACE_PROFILE_ADDRESS.getRoute(activePolicyID, Navigation.getActiveRoute()));
```

Comment:
Same, this should be happening in the view.

Reply:
Created a follow-up issue for this.